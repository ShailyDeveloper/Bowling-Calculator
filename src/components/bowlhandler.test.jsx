import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BowlHandler from './bowlhandler';
import BowlValue from '../ui/bowlingfields';
import BowlButton from '../ui/bowlingbutton';
import '@testing-library/jest-dom';

configure({ adapter: new Adapter() });

describe('Bowl Handler', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4" }} />);
    });
    it('match the snapshot of BowlHandler function', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('BowlValues should match its snapshot', () => {
        expect(BowlValue).toMatchSnapshot();
    });
    it('BowlButton should match its snapshot', () => {
        expect(BowlButton).toMatchSnapshot();
    });
    it('BowlValues should be rendered twice when the showval is true', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", showval: true }} />);
        expect(wrapper.find(BowlValue)).toHaveLength(2);
    });
    it('BowlValues should not be rendered once when the showval is true', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", showval: true }} />);
        expect(wrapper.find(BowlValue)).not.toHaveLength(1);
    });
    it('BowlValues should be rendered once when the showval is false', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", showval: false }} />);
        expect(wrapper.find(BowlValue)).toHaveLength(1);
    });
    it('BowlValues should not be rendered twice when the showval is false', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", showval: false }} />);
        expect(wrapper.find(BowlValue)).not.toHaveLength(2);
    });
    it('BowlButton should not render for first strike', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "10", laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(0);
    });
    it('BowlButton should render once for first strike', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "10", laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for second strike', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str2Val: "10", laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(0);
    });
    it('BowlButton should render once for second strike', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str2Val: "10", laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for second strike', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str2Val: "10", laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should render once for a spare', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "4", lastspare: 0 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for a spare', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "4", lastspare: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should render once for a normal round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 0, laststrike: 0 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for a normal round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 1, laststrike: 0 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for a normal round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 1, laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for a normal round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 0, laststrike: 1 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render for first strike in the final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "10", laststrike: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(0);
    });
    it('BowlButton should render once for first strike in the final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "10", laststrike: 0, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for second strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str2Val: "10", laststrike: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(0);
    });
    it('BowlButton should render once for second strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str2Val: "10", laststrike: 0, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for a spare in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", str2Val: "5", lastspare: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should render once for a spare in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "4", str2Val: "5", lastspare: 0, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for a spare in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "4", lastspare: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should render once for a normal strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 0, laststrike: 0, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).toHaveLength(1);
    });
    it('BowlButton should not render once for a normal strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 1, laststrike: 0, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for a normal strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 1, laststrike: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });
    it('BowlButton should not render once for a normal strike in final round', () => {
        wrapper = shallow(<BowlHandler showsecond={{ str1Val: "6", str2Val: "2", lastspare: 0, laststrike: 1, roundcount: 10 }} />);
        expect(wrapper.find(BowlButton)).not.toHaveLength(1);
    });

});
