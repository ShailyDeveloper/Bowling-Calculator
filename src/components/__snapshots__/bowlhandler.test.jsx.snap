// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bowl Handler BowlButton should match its snapshot 1`] = `[Function]`;

exports[`Bowl Handler BowlValues should match its snapshot 1`] = `[Function]`;

exports[`Bowl Handler match the snapshot of BowlHandler function 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BowlHeaders
    showsecond={
        Object {
            "str1Val": "4",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <b
          style={
                    Object {
                              "margin": "10px",
                            }
          }
>
          Round 
</b>,
        <br />,
        <br />,
        <Bowlval
          elementval={
                    <option
                              value="4"
                    >
                              4
                    </option>
          }
          handleChange={undefined}
          strikecount="1"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Round ",
            undefined,
          ],
          "style": Object {
            "margin": "10px",
          },
        },
        "ref": null,
        "rendered": Array [
          "Round ",
          undefined,
        ],
        "type": "b",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "elementval": <option
            value="4"
>
            4
</option>,
          "handleChange": undefined,
          "strikecount": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <b
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            Round 
</b>,
          <br />,
          <br />,
          <Bowlval
            elementval={
                        <option
                                    value="4"
                        >
                                    4
                        </option>
            }
            handleChange={undefined}
            strikecount="1"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Round ",
              undefined,
            ],
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": Array [
            "Round ",
            undefined,
          ],
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "elementval": <option
              value="4"
>
              4
</option>,
            "handleChange": undefined,
            "strikecount": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
